react-state/my-app/src/ToggleButton.tsx

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+
+type Props = {
+  text: string;
+  color: string;
+};
+
+export function ToggleButton({ text, color }: Props) {
+  const [isClicked, setIsClicked] = useState(false);
+  console.log('value returned by useState', isClicked);
+
+  function handleClick() {
+    console.log('value before calling setter:', isClicked);
+    setIsClicked(!isClicked);
+    console.log('value after calling setter:', isClicked);
+  }
+
+  if (isClicked) {
+    return (
+      <button style={{ backgroundColor: color }} onClick={handleClick}>
+        {text}
+      </button>
+    );
+  }
+  return (
+    <button style={{ backgroundColor: 'gray' }} onClick={handleClick}>
+      {text}
+    </button>
+  );
+}
```

Comment:
It works with these separate return statements, but there's lots of duplicated code. The only difference between the two blocks is the background color. Can you figure out how to eliminate all that duplicate code?